src/bjointsp/template/component.py

```diff
@@ -1,5 +1,5 @@
 class Component:
-    def __init__(self, name, type, stateful, inputs, outputs, cpu, mem, dr, config=None):
+    def __init__(self, name, type, stateful, inputs, outputs, cpu, mem, dr, vnf_delay, config=None):
```

Comment:
`vnf_delay` should be optional. `vnf_delay=None`; check if it's assigned otherwise set to 0 (and write debug logging comment).

Reply:
In your example, the total delay should be 105, but it is calculated (incorrectly) as 140 (because 35 is added twice).

Reply:
The default is already set in the reader file, so if it doesn't find a vnf_delay in the yaml file it will default to zero. I will add a default in the function definition as well.